### PR TITLE
Increase JET failure count for julia 1.11

### PR DESCRIPTION
**Pull request by lgoettgens**
In julia 1.11, there are some more issues reported by JET. This PR increases the number again such that the CI job succeeds again.
IMO, this doesn't need a changelog entry.

- [x] The code is properly formatted and commented.
- [x] Substantial new functionality is documented within the docs.
- [x] All new functionality is tested.
- [x] All of the automated tests on github pass.
- [x] We recently started enforcing formatting checks. If formatting issues are reported in the new code you have written, please correct them.

ping @Krastanov

### CODE REVIEW

File: test/test_jet.jl [hunk 1]
```diff
@@ -24,6 +24,7 @@
     ))
 
     @show rep
+    @show length(JET.get_reports(rep))
     @test_broken length(JET.get_reports(rep)) == 0
-    @test length(JET.get_reports(rep)) <= 10
+    @test length(JET.get_reports(rep)) <= 22
```
Comment (lgoettgens) [inline test/test_jet.jl:29]:
```suggestion
    @test length(JET.get_reports(rep)) <= 23
```
Hmm, CI gets one more report than I get locally. Let me just increase all numbers by one for now, and I'll look into the exact issue later.

**Comment by Fe-r-oz** [comment]
Hi, Lars,  If you are wondering about that 1 extra error - That one error is from `quantum_mallows` which is not reported locally, the last time I checked.  Please do check whether this is reported locally,  I don't think so.

```
┌ quantum_mallows(rng::Any, n::Any) @ QuantumClifford /home/runner/work/QuantumClifford.jl/QuantumClifford.jl/src/randoms.jl:198
│┌ popat!(x::PythonCall.Wrap.PyList{T}, i::Integer) where T @ PythonCall.Wrap /home/runner/.julia/packages/PythonCall/Nr75f/src/Wrap/PyList.jl:76
││┌ pyconvert(::Type{T}, x::PythonCall.Core.Py) where T @ PythonCall.Convert /home/runner/.julia/packages/PythonCall/Nr75f/src/Convert/pyconvert.jl:382
│││┌ pyconvert_rule_range(::Type{…}, x::PythonCall.Core.Py, ::Type{…}, ::Type{…}) where {…} @ PythonCall.Convert /home/runner/.julia/packages/PythonCall/Nr75f/src/Convert/rules.jl:159
││││┌ (::Type{StepRange{_A, _B}} where {_A, _B})(start::Any, step::Any, stop::Any) @ Base ./range.jl:329
│││││┌ steprange_last(start::Any, step::Any, stop::Any) @ Base ./range.jl:358
││││││┌ (::Base.var"#overflow_case#86")(absdiff::Signed, absstep::Any) @ Base ./range.jl:357
│││││││┌ unsigned(x::Union{Int128, Int16, Int32, Int64, Int8}) @ Base ./int.jl:216
││││││││┌ reinterpret(::Union{Type{…}, Type{…}, Type{…}, Type{…}, Type{…}}, x::Union{Int128, Int16, Int32, Int64, Int8}) @ Base ./essentials.jl:729
│││││││││ bitcast: target type not a leaf primitive type: Base.bitcast(Out::Union{Type{UInt128}, Type{UInt16}, Type{UInt32}, Type{UInt64}, Type{UInt8}}, x::Union{Int128, Int16, Int32, Int64, Int8})
││││││││└────────────────────
```

Previously, even this was not reported locally, but now, maybe `fast_rows` now is since there is only 1 error that is not reported locally:
```

 fastrow(t::QuantumClifford.Tableau{Tₚᵥ, Tₘ}) where {Tₚᵥ, Tₘ<:LinearAlgebra.Adjoint} @ QuantumClifford /home/runner/work/QuantumClifford.jl/QuantumClifford.jl/src/fastmemlayout.jl:19
│┌ collect(A::LinearAlgebra.Adjoint) @ Base ./array.jl:707
││┌ _collect_indices(::Tuple{}, A::LinearAlgebra.Adjoint) @ Base ./array.jl:722
│││┌ copyto!(dest::Array{_A, 0} where _A, src::LinearAlgebra.Adjoint) @ Base ./abstractarray.jl:1055
││││┌ isempty(a::LinearAlgebra.Adjoint) @ Base ./abstractarray.jl:1212
│││││┌ length(a::LinearAlgebra.Adjoint{T, S} where {T, S<:Union{…}}) @ StaticArrays /home/runner/.julia/packages/StaticArrays/MSJcA/src/abstractarray.jl:1
││││││┌ StaticArraysCore.Size(a::LinearAlgebra.Adjoint{T, S} where {T, S<:Union{…}}) @ StaticArraysCore /home/runner/.julia/packages/StaticArraysCore/7xxEJ/src/StaticArraysCore.jl:515
│││││││┌ StaticArraysCore.Size(::Type{Union{}}) @ StaticArraysCore /home/runner/.julia/packages/StaticArraysCore/7xxEJ/src/StaticArraysCore.jl:517
││││││││ `S` not defined in static parameter matching: S::Type{S} where S<:Tuple
│││││││└────────────────────
```

If these two don't help, then you have to check the log https://github.com/QuantumSavory/QuantumClifford.jl/actions/runs/11257640539/job/31302368032#step:6:2408 to see which one is not reported. I hope this helps!

**Comment by lgoettgens** [comment]
Thanks @Fe-r-oz for the heads-up. comparing my local log with the CI log it seems to be the `quantum_mallows` one.
My plan is to first fix all other things, and then look into this one last. Maybe this can be condensed to an mwe for a JET issue.

**Comment by Krastanov** [comment]
Thank you!

**Comment by lgoettgens** [comment]
I have some more information about this `quantum_mallow` failure that only happens in CI (in case you are interested @Fe-r-oz):
If you run the complete testsuite locally (i.e. `include("test/runtests.jl")`), it appears as well, and just running the JET tests afterwards will report it as well. So this seems to be somehow dependent on some testfile that runs before.
I currently think this is due to some Python/PythonCall shenanigans, but I'll need to dig a bit deeper.

**Pull request merged**